### PR TITLE
LPS-158736 | Expose ERC in Structured Content Folders

### PR DESCRIPTION
**Pull request by Hijky**
**Background**
Add test to assert Structured Content Folder is created with default ERC
Add test to assert Structured Content Folder is created with custom ERC
Add test to assert Not existed Structured Content Folder will be created with custom ERC
Add test to assert Existed Structured Content Folder will be updated with custom ERC
Add test to assert Can get correct details with default erc from folders
Add test to assert Can get correct details with custom erc from folders
Add test to assert Structured Content Folder Is Deleted With Default ERC
Add test to assert Structured Content Folder Is Deleted With Custom ERC

**Test Plan**
- LPS-158883 
Given a Site
When with postSiteStructuredContentFolder
Then a new folder is created with the erc value equals to the uuid in the body response
- LPS-158902
Given a Site
When with postSiteStructuredContentFolder
Then a new folder is created with the custom erc in the body response
- LPS-158903
Given a Site
When with putSiteStructuredContentFolderByExternalReferenceCode 
Then a new folder is created with custom erc in the body response
- LPS-158904
Given a Site
Given a folder with custom erc
When with putSiteStructuredContentFolderByExternalReferenceCode to update folder name
Then the same folder's name is updated in a response body of getStructureContentFolderById
- LPS-158905
Given a Site
Given a folder
When I use getSiteStructuredContentFolderByExternalReferenceCode
Then I can retrieve the correct information from the response
- LPS-158907
Given a Site 
Given a folder with custom erc
When I use getSiteStructuredContentFolderByExternalReferenceCode
Then I can retrieve the correct information from the response
- LPS-158908
Given a Site
Given a folder
When I use deleteSiteStructuredContentFolderByExternalReferenceCode
Then the folder is deleted successfully
- LPS-158909
Given a Site
Given a folder with custom erc
When I use deleteSiteStructuredContentFolderByExternalReferenceCode
Then the folder is deleted successfully

**JIRA Ticket**
- https://issues.liferay.com/browse/LPS-158736

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by Hijky** [comment]
Failed test are unrelated to this pr, failed tests are [LPS-158454](https://issues.liferay.com/browse/LPS-158454):

AdvancedObjectAPI#CanReturnAggregationTermsDetailsInOneToManyRelationship
AdvancedObjectAPI#CanReturnAggregationTermsDetailsInOneToManyRelationshipAfterObjectDeletion
AdvancedObjectAPI#CanReturnAggregationTermsDetailsInOneToManyUpdatedRelationship

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeERCInStructuredContentFolder.testcase [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		task ("Then the folder is deleted successfully") {
+			HeadlessWebcontentFolderAPI.assertStructuredContentFolderHasDeleted(
+				expectedValue = "No JournalFolder exists with the primary key ${structuredContentFolderId}",
+				structuredContentFolderId = "${structuredContentFolderId}");
+		}
+	}
+
+	@description = "Structured Content Folder Is Deleted With Default ERC"
+	@priority = "5"
+	test StructuredContentFolderIsDeletedWithDefaultErc {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeERCInStructuredContentFolder.testcase:199]: `CanDeleteStructuredContentFolderByErc`

**Review by magjed4289 — APPROVED** [review]

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**